packages/opentelemetry-instrumentation-anthropic/opentelemetry/instrumentation/anthropic/streaming.py

```diff
@@ -135,30 +157,26 @@ def build_from_streaming_response(
     # calculate token usage
     if Config.enrich_token_usage:
         try:
-            prompt_tokens = -1
             completion_tokens = -1
-
             # prompt_usage
-            if kwargs.get("prompt"):
-                prompt_tokens = instance.count_tokens(kwargs.get("prompt"))
-            elif kwargs.get("messages"):
-                prompt_tokens = sum(
-                    [
-                        instance.count_tokens(m.get("content"))
-                        for m in kwargs.get("messages")
-                    ]
-                )
+            if usage := complete_response.get("usage"):
+                prompt_tokens = usage.get("input_tokens", 0)
+            else:
+                prompt_tokens = count_prompt_tokens_from_request(instance, kwargs)
 
             # completion_usage
-            completion_content = ""
-            if complete_response.get("events"):
-                model_name = complete_response.get("model") or None
-                for event in complete_response.get("events"):  # type: dict
-                    if event.get("text"):
-                        completion_content += event.get("text")
-
-                if model_name:
-                    completion_tokens = instance.count_tokens(completion_content)
+            if usage := complete_response.get("usage"):
+                completion_tokens = usage.get("output_tokens", 0)
+            else:
+                completion_content = ""
+                if complete_response.get("events"):
+                    model_name = complete_response.get("model") or None
+                    for event in complete_response.get("events"):  # type: dict
+                        if event.get("text"):
+                            completion_content += event.get("text")
+
+                    if model_name:
+                        completion_tokens = instance.count_tokens(completion_content)
```

Comment:
In both these places (same in async below), I default to getting the input and output tokens from the response, because I think it is:
- a better source of truth
- more efficient

This also caused me to change some values and fixtures in test, where the estimate from `count_tokens` was way off – 17 instead of 8.

Let me know, if this was a design decision to default to counting tokens, and if I have missed something

Reply:
Nope, that's actually better! Anthropic used to not return the token count for streaming requests